sdk/src/main/java/io/dapr/client/DaprClientHttp.java

```diff
@@ -763,15 +766,134 @@ private <T> QueryStateResponse<T> buildQueryStateResponse(DaprHttp.Response resp
+      queryParams.put("key", Collections.unmodifiableList(keys));
+
+      return Mono.subscriberContext().flatMap(
+              context -> this.client
+                      .invokeApi(
+                              DaprHttp.HttpMethods.GET.name(),
+                              pathSegments, queryParams,
+                              (String) null, null, context)
+      ).map(
+              response -> {
+                try {
+                  JsonNode root = INTERNAL_SERIALIZER.parseNode(response.getBody());
+                  List<ConfigurationItem> result = new ArrayList<>();
+                  for (Iterator<JsonNode> it = root.elements(); it.hasNext(); ) {
+                    JsonNode node = it.next();
+                    String key = node.path("key").asText();
+                    String value = node.path("value").asText();
+                    String version = node.path("version").asText();
+                    ConfigurationItem configurationItem = new ConfigurationItem(key, value, version);
+                    result.add(configurationItem);
+                  }
+                  return result;
+                } catch (IOException e) {
+                  throw new RuntimeException(e);
+                }
+              }
+      );
+    } catch (Exception ex) {
+      return DaprException.wrapMono(ex);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Flux<SubscribeConfigurationResponse> subscribeConfiguration(SubscribeConfigurationRequest request) {
+    try {
+      final String configurationStoreName = request.getStoreName();
+      final List<String> keys = request.getKeys();
+
+      if (configurationStoreName == null || (configurationStoreName.trim().isEmpty())) {
+        throw new IllegalArgumentException("Configuration Store Name can not be null or empty.");
+      }
+      if (keys.isEmpty()) {
+        throw new IllegalArgumentException("Keys can not be null or empty.");
+      }
+
+      String[] pathSegments =
+              new String[] { DaprHttp.ALPHA_1_API_VERSION, "configuration", configurationStoreName, "subscribe" };
+      Map<String, List<String>> queryParams = new HashMap<>();
+      queryParams.put("key", Collections.unmodifiableList(keys));
+      SubscribeConfigurationResponse res = Mono.subscriberContext().flatMap(
+              context -> this.client.invokeApi(
+                      DaprHttp.HttpMethods.GET.name(),
+                      pathSegments, queryParams,
+                      (String) null, null, context
+              )
+      ).map(response -> {
+        try {
+          JsonNode root = INTERNAL_SERIALIZER.parseNode(response.getBody());
+          String subscriptionId = root.path("id").asText();
+          return new SubscribeConfigurationResponse(subscriptionId, new ArrayList<>());
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+      }).block();
+      if (res != null) {
+        return Flux.just(res);
```

Comment:
Here I could not think of better way of handelling this.
Reason -  We have common interface for http and gRPC. The gRPC api is implemented using stream data type in runtime. So, here we need to have Flux as return type but HTTP API is implemented differently in runtime which just returns a subscription id and then dapr makes a callback to app endpoint for getting the subscribed changes. So here Mono return type is appropriate. Thus the above logic of returning just one item as Flux.

Reply:
looks good ... @artursouza I think this is good enough for now ... any concerns?

Reply:
I am fine with this.